FindxDAQ.cmake

```diff
@@ -13,245 +14,312 @@ set(xDAQ_FOUND TRUE)
+        endif()
+        unset(msg_type)
     endif()
+    unset(found)
 endforeach()
 
-# Check for threading libraries only if required
-set(xdaq_need_threads FALSE)
+# Build a list of all requested libraries with dependencies included
+set(xdaq_requested_libs ${xDAQ_FIND_COMPONENTS})
+foreach(lib ${xDAQ_FIND_COMPONENTS})
+    list(APPEND xdaq_requested_libs ${xdaq_${lib}_recursive_depends})
+endforeach()
+list(REMOVE_DUPLICATES xdaq_requested_libs)
 
-# Add dependencies to the list of libs
-set(xdaq_required_libs "")
+# Turn the recursive list of dependencies into a list of required variables
 foreach(lib ${xDAQ_FIND_COMPONENTS})
-    list(APPEND xdaq_required_libs ${lib})
-    list(APPEND xdaq_required_libs ${xdaq_${lib}_depends})
+    set(xdaq_${lib}_required_variables xDAQ_${lib}_LIBRARY xDAQ_INCLUDE_DIRS)
+    foreach(dep ${xdaq_${lib}_recursive_depends})
+        list(APPEND xdaq_${lib}_required_variables xDAQ_${dep}_LIBRARY)
 
-    if(xdaq_${name}_threads)
+        # Threads
+        if(xdaq_${dep}_threads)
+            list(APPEND xdaq_${lib}_required_variables CMAKE_THREAD_LIBS_INIT)
+        endif()
+
+        # Toolbox requires libuuid from the system
+        if(${dep} STREQUAL "toolbox")
+            list(APPEND xdaq_${lib}_required_variables xDAQ_uuid_LIBRARY)
+        endif()
+    endforeach()
+
+    # Threads
+    if(xdaq_${lib}_threads)
+        list(APPEND xdaq_${lib}_required_variables CMAKE_THREAD_LIBS_INIT)
+    endif()
+
+    # Toolbox requires libuuid from the system
+    if(${lib} STREQUAL "toolbox")
+        list(APPEND xdaq_${lib}_required_variables xDAQ_uuid_LIBRARY)
+    endif()
+
+    list(REMOVE_DUPLICATES xdaq_${lib}_required_variables)
+endforeach()
+
+# Are threads required?
+foreach(lib ${xdaq_requested_libs})
+    if(xdaq_${lib}_threads)
         set(xdaq_need_threads TRUE)
     endif()
 endforeach()
-list(REMOVE_DUPLICATES xdaq_required_libs)
 
-# Threads
+# Find threads
 if(xdaq_need_threads)
     find_package(Threads QUIET)
 endif()
 
-# Creates an imported target for the given lib
-function(_xdaq_import_lib name)
-    string(TOUPPER ${name} uname)
-    set(xDAQ_${uname}_FOUND TRUE PARENT_SCOPE)
-
-    # Try to find the library
-    find_library(
-        xDAQ_${uname}_LIBRARY
-        ${name}
-        NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH
-        HINTS ENV XDAQ_ROOT
-        PATHS /opt/xdaq/
-        PATH_SUFFIXES lib lib64
-        DOC "Root directory of the xDAQ installation")
-
-    mark_as_advanced(xDAQ_${uname}_LIBRARY)
-
-    if(NOT xDAQ_${uname}_LIBRARY)
-        set(xDAQ_FOUND FALSE PARENT_SCOPE)
-        set(xDAQ_${uname}_FOUND FALSE PARENT_SCOPE)
-
-        if(xDAQ_FIND_REQUIRED)
-            message(SEND_ERROR "Could not find shared object file for xDAQ library ${name}")
-        endif()
-        # Do not create the target
-        return()
+# Sets xDAQ_FOUND to FALSE if the given library isn't optional
+function(_xdaq_lib_not_found lib)
```

Comment:
This function is not used anywhere. I guess it is a leftover and is advantageously replaced by `find_package_handle_standard_args`, line 299 ?

Reply:
Also a good catch :)